include/rmm/exec_policy.hpp

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ @file exec_policy.hpp
+ Thrust execution policy that uses RMM's Thrust Allocator Adaptor.
+ */
+
+#pragma once
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/mr/device/thrust_allocator_adaptor.hpp>
+
+namespace rmm {
+
+/**
+ * @brief Returns a Thrust CUDA execution policy that uses RMM for temporary memory allocation on
+ * the specified stream.
+ */
+inline auto exec_policy(cuda_stream_view stream = cuda_stream_default)
+{
+  return thrust::cuda::par(rmm::mr::thrust_allocator<char>(stream)).on(stream.value());
```

Comment:
I'm thinking while we're updating this it would make sense to add a second parameter for the MR passed to the `thrust_allocator` (defaulted to `get_current_device_resource`).

Reply:
OK. I assume we did not do that before because non-default MR's are only used in RAPIDS libs for outputs, and any allocation done using the exec_policy is by definition temporary.

Reply:
Right, I don't expect we'll make use of this change in RAPIDS libs anytime soon, but it's a simple enough thing that other users may want to take advantage of.

Reply:
Done.